gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -251,6 +251,34 @@ case class TransformPreOverrides(isAdaptiveContext: Boolean)
     }
   }
 
+  def applyScanNotTransformable(plan: SparkPlan): SparkPlan = plan match {
+    case plan: FileSourceScanExec =>
+      val newPartitionFilters =
+        ExpressionConverter.transformDynamicPruningExpr(plan.partitionFilters, reuseSubquery)
+      val newSource = plan.copy(partitionFilters = newPartitionFilters)
+      TransformHints.tagNotTransformable(newSource, "fallback file source scan")
```

Comment:
There are two minor tests failed related to unexpected fallback reason. Maybe, we need to re-use the tag of `plan` for `newSource` to keep the original fallback reason. e.g.,

`TransformHints.tag(newSource, TransformHints.getHint(plan))`